torch/csrc/jit/script/class_type.cpp

```diff
@@ -8,7 +8,8 @@ namespace c10 {
 // c10. Sigh...
 
 Function* ClassType::getMethod(const std::string& name) const {
-  return compilation_unit_->find_function(name);
+  const auto qualname = QualifiedName(*qualified_name_obj(), name);
```

Comment:
qualified_name_obj is really confusing. If you look at the methods in NamedType they are pretty redundant. We really should have NamedType just return the qualified name and rename qualified_name_obj to just qualifiedName.